src/scenes/wizard/index.ts

```diff
@@ -33,10 +33,11 @@ export class WizardScene<C extends WizardContext = WizardContext>
   }
 
   middleware() {
-    return Composer.compose<C, WizardContext>([
+    return Composer.compose<C>([
       (ctx, next) => {
-        const wizard = new WizardContextWizard<C>(ctx, this.steps)
-        return next(Object.assign(ctx, { wizard }))
+        const steps = this.steps as Array<MiddlewareFn<WizardContext>>
+        ctx.wizard = new WizardContextWizard<WizardContext>(ctx, steps)
```

Comment:
Why is the cast necessary?

Reply:
For the same reason as here:
https://github.com/telegraf/telegraf/pull/1251/files#diff-e6b547a30db4384fb39b5ef00d301d72dc76758ce50a4a5c1c216820c73cb863R35-R38

Just that we cannot cast there, while this is possible here.

Reply:
Alright. Comment would be nice. LGTM other than that.